src/GraphQL.Tests/DocumentWriterTests.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using GraphQL.SystemTextJson;
+using Xunit;
+
+namespace GraphQL.Tests
+{
+    /// <summary>
+    /// Tests for <see cref="IDocumentWriter"/> implementations and the custom converters
+    /// that are used in the process of serializing an <see cref="ExecutionResult"/> to JSON.
+    /// </summary>
+    public class DocumentWriterTests
+    {
+        [Theory]
+        [ClassData(typeof(DocumentWritersTestData))]
+        public async void Can_Write_Execution_Result(IDocumentWriter writer)
```

Comment:
Thank you for these tests. 🥇

Reply:
There's probably more that can be written, I didn't nitpick the serializers, for instance I think there's something custom for apollo trace, like it always need to be camelcase, but... it's a start.